src/reanimated2/WorkletEventHandler.ts

```diff
@@ -1,13 +1,93 @@
 'use strict';
 import type { NativeSyntheticEvent } from 'react-native';
 import { registerEventHandler, unregisterEventHandler } from './core';
-import type { EventPayload, ReanimatedEvent } from './hook/commonTypes';
+import type {
+  EventPayload,
+  ReanimatedEvent,
+  IWorkletEventHandler,
+} from './hook/commonTypes';
 import { shouldBeUseWeb } from './PlatformChecker';
 
 const SHOULD_BE_USE_WEB = shouldBeUseWeb();
 
-type JSEvent<Event extends object> = NativeSyntheticEvent<EventPayload<Event>>;
+type EventHandlerRegistration = {
+  id: number;
+  viewTag: number;
+};
+
+class WorkletEventHandlerNative<Event extends object>
+  implements IWorkletEventHandler<Event>
+{
+  worklet: (event: ReanimatedEvent<Event>) => void;
+  eventNames: string[];
+  viewTags: Set<number>;
+  registrations: EventHandlerRegistration[];
+  constructor(
+    worklet: (event: ReanimatedEvent<Event>) => void,
+    eventNames: string[] = []
+  ) {
+    this.worklet = worklet;
+    this.eventNames = eventNames;
+    this.viewTags = new Set<number>();
+    this.registrations = [];
+  }
+
+  updateEventHandler(
+    newWorklet: (event: ReanimatedEvent<Event>) => void,
+    newEvents: string[]
+  ): void {
+    // Update worklet and event names
+    this.worklet = newWorklet;
+    this.eventNames = newEvents;
+
+    // Detach all events
+    this.registrations.forEach((registration) => {
+      unregisterEventHandler(registration.id);
+    });
+    this.registrations = [];
+
+    // Attach new events with new worklet
+    this.registrations = this.eventNames.flatMap((eventName) => {
+      return Array.from(this.viewTags).map((tag) => {
+        return {
+          id: registerEventHandler(this.worklet, eventName, tag),
+          viewTag: tag,
+        };
+      });
+    });
```

Comment:
I wonder if we actually want to have it as a flattened array 🤔

Reply:
@tjzel You might have a point, the only time `registrations` are read is when we want to unregister some events for a specific viewTag. I will try to make it into a map with viewTag keys and an array of corresponding registrationID's.